crates/bevy_transform/src/systems.rs

```diff
@@ -1,75 +1,84 @@
 use crate::components::{GlobalTransform, Transform};
-use bevy_ecs::prelude::{Changed, Entity, Query, With, Without};
+use bevy_ecs::prelude::{Changed, Entity, Or, Query, With, Without};
 use bevy_hierarchy::{Children, Parent};
 
+/// Update [`GlobalTransform`] component of entities that aren't in the hierarchy
+pub fn transform_propagate_system_flat(
+    mut query: Query<
+        (&Transform, &mut GlobalTransform),
+        (Changed<Transform>, Without<Parent>, Without<Children>),
+    >,
+) {
+    for (transform, mut global_transform) in query.iter_mut() {
+        *global_transform = GlobalTransform::from(*transform);
```

Comment:
This isn't a problem with this PR, but it's an interesting conundrum:

Currently, if the `GlobalTransform` has been erroneously changed by users to something other than the correct transform, that will remain until something higher up in the hierarchy or the `Transform` changes. Do we want this behaviour, or do we want to fix this (or warn?)

Reply:
Simple solution: make `GlobalTransform` read-only the same way #4197 does with the hierarchy itself. Makes this footgun a lot harder to come across.